Adding parent document retrieval in default RAG pipeline

star-nox opened the pull request.

rohan-uiuc reviewed (commented):
Needs some performance improvements
* Extract all sql documents in a single db call moving it out of the executor. 
* Remove redundant formatting of types by looping over all docs. 2 sequential and 1 parallel loop can be reduced to a single parallel loop. 
* Let's compare the API response time with current code and suggested improvements.

star-nox commented:
Changes made:
1. Removed redundant call to `format_for_json_mqr()` at the end of `getTopContexts()`. The contexts are already formatted in context padding.
2. Querying SQL data for top 5 contexts at once (making queries based on either urls or s3_paths) and supplying it to `supabase_context_padding`. Now performing 2 SQL queries (worst case) instead of 5.
3. Modified padding functions to remove key checks in dictionaries.

Total context retrieval time ranges from 2.5s to 4s
